diktat-rules/src/main/kotlin/com/saveourtool/diktat/ruleset/utils/AstNodeUtils.kt

```diff
@@ -1150,6 +1173,40 @@ fun doesLambdaContainIt(lambdaNode: ASTNode): Boolean {
     return hasNoParameters(lambdaNode) && hasIt
 }
 
+/**
+ * Checks that node (property or backing field) has pair node (backing field or property)
+ *
+ * @return true if node has a pair
+ */
+@Suppress("CyclomaticComplexMethod")
+internal fun isPairPropertyBackingField(propertyNode: ASTNode?, backingFieldNode: ASTNode?): Boolean {
```

Comment:
Maybe split this method into 2 different methods (smth like `isPropertyHasBackingField` and `isBackingFieldForExistingProperty`) to avoid such calls `isPairPropertyBackingField(node.treeParent, null)` and `isPairPropertyBackingField(null, node)`. Аfter that you can try to extract the common code into a separate method.

Reply:
Discussed offline: don't need it.